infra/modules/api/main.tf

```diff
@@ -0,0 +1,48 @@
+resource "random_password" "jwt_secret" {
+  length  = 256
+  keepers = {
+    pass_version = 1
+  }
+}
+
+resource "aws_ssm_parameter" "jwt_secret" {
+  name  = "jwt_secret_${var.environment}_${var.project_name}"
+  type  = "String"
+  value = random_password.jwt_secret.result
+}
+
+data "aws_ssm_parameter" "orcid_app_id" {
+  name  = "orcid_app_id_${var.environment}_${var.project_name}"
+  // Must be manually created before terraform run. It is required for API releases.
+  // The orcid api client id, obtained from orcid.
+}
+
+data "aws_ssm_parameter" "orcid_secret_key" {
+  name  = "orcid_secret_key_${var.environment}_${var.project_name}"
+  // Must be manually created before terraform run. It is required for API releases.
+  // The orcid api client secret, obtained from orcid."
+}
+
+data "aws_ssm_parameter" "doi_prefix" {
+  name  = "doi_prefix_${var.environment}_${var.project_name}"
+  // Must be manually created before terraform run. It is required for API releases.
+  // The DOI prefix used for minting datacite DOIs.
+}
+
+data "aws_ssm_parameter" "datacite_endpoint" {
+  name  = "datacite_endpoint_${var.environment}_${var.project_name}"
+  // Must be manually created before terraform run. It is required for API releases.
+  // The datacite endpoint used for minting DOIs.
+}
+
+data "aws_ssm_parameter" "datacite_user" {
+  name  = "datacite_user_${var.environment}_${var.project_name}"
+  // Must be manually created before terraform run. It is required for API releases.
+  // The username of the datacite user, used for minting DOIs.
+}
+
+data "aws_ssm_parameter" "datacite_password" {
+  name  = "datacite_password_${var.environment}_${var.project_name}"
+  // Must be manually created before terraform run. It is required for API releases.
+  // The password of the datacite user, used for minting DOIs.
+}
```

Comment:
The other environment variables added as `data` blocks. I think this is worth discussing on if we want to do this.

These data blocks make the terraform fail if the parameters don't exist in parameter store. This moves the failure from happening at API deploy time, back to infrastructure creation time. I think it also documents what's required.

However, this might not be where we want the failure to happen. Perhaps it just needs a multi-line comment in this file?